### ISSUE TITLE
assign list.sort() instead of sorted(list)

### ISSUE THREAD

**Issue by orSolocate** [PyCQA/pylint#5722]
### Current problem

when you use the `sort()` method on a list object, it modifies the list and returns None.
A common mistake is to assign the return value of this assignment (i.e. None) to a variable thinking you assigned the sorted list instead.

```
lst=[3,2,1]
A=lst.sort()
```

Results with: 
None in `A` and [1,2,3] in `lst`.

### Desired solution

The checker should raise the WARNING message:
“A sort method of `list` object is assigned to a variable and equals None, consider using the `sorted()` function instead”.

### Additional context

When i run pylint with all extensions on the example above I don't get any message.
But PyCharm's linter has a message for it so I am not sure if it was already implemented.

If not I could write it.

**Comment by DanielNoord** [comment]
I like the idea although I wonder how opinionated this might be. Personally I never use `.sort()` to return `None`, but others might actually rely on that behaviour.

@Pierre-Sassoulas Should this be a plugin or a standard check?

**Comment by Pierre-Sassoulas** [comment]
I also like this. Don't we have 'assignment-from-none' already? Is it a false negative?

**Comment by DanielNoord** [comment]
Yes, I think we can consider this a false-positive.

**Comment by orSolocate** [comment]
In the `list.sort()` method documentation they write:

`        Sort the list in ascending order and return None.`

So I assume it always returns `None`.. therefore I don't see a case when someone relies on this value.
If they did, they should assign None to the variable/expression they want to evaluate..

@Pierre-Sassoulas  I could try to expand the 'assignment-from-none' checker...
